pkg/storage/registry/spaces/spaces.go

```diff
@@ -98,27 +103,35 @@ func (c *config) init() {
 	if len(c.Providers) == 0 {
 		c.Providers = map[string]*provider{
 			sharedconf.GetGatewaySVC(""): {
-				MountPath: "/",
+				Spaces: map[string]*spaceConfig{
+					"personal": {MountPoint: "/users", PathTemplate: "/users/{{.Space.Owner.Id.OpaqueId}}"},
+					"project":  {MountPoint: "/projects", PathTemplate: "/projects/{{.Space.Name}}"},
+					"share":    {MountPoint: "/users/{{.CurrentUser.Id.OpaqueId}}/Shares", PathTemplate: "/users/{{.CurrentUser.Id.OpaqueId}}/Shares/{{.Space.Name}}"},
+					"public":   {MountPoint: "/public"},
+				},
 			},
 		}
 	}
 
-	// cleanup rule paths
-	for _, rule := range c.Providers {
-		// if the path template is not explicitly set use the mountpath as path template
-		if rule.PathTemplate == "" && strings.HasPrefix(rule.MountPath, "/") {
-			// TODO err if the path is a regex
-			rule.PathTemplate = rule.MountPath
-		}
+	// cleanup space paths
+	for _, provider := range c.Providers {
+		for _, space := range provider.Spaces {
 
-		// cleanup path template
-		rule.PathTemplate = filepath.Clean(rule.PathTemplate)
+			// if the path template is not explicitly set use the mountpath as path template
+			if space.PathTemplate == "" && strings.HasPrefix(space.MountPoint, "/") {
```

Comment:
1) Should we error if both `space.PathTemplate` and `space.MountPoint` are empty strings?

2) Why does `space.MountPoint` need to start with a `"/"`? Isn't that added automatically later?

Reply:
1. No, mount point and path template should default to `/`
2. The check makes sure it is not a regex, starting with `^` ... yes, that only works with a ton of assumptions needs better test coverage to explain how regexes are used

Reply:
I don't quite understand 2. If it is a regex, why can't it start with `"/"`?